app/models/provider_interface/provider_applications_filter.rb

```diff
@@ -66,12 +66,16 @@ def search_filter
     end
 
     def recruitment_cycle_filter
-      cycle_options = RecruitmentCycle::CYCLES.map do |year, label|
-        {
-          value: year,
-          label: label,
-          checked: applied_filters[:recruitment_cycle_year]&.include?(year),
-        }
+      cycle_options = RecruitmentCycle::CYCLES
+        .select do |year, _l|
+          RecruitmentCycle.years_visible_to_providers.map(&:to_s).include? year
+        end
+        .map do |year, label|
+          {
+            value: year,
+            label: label,
+            checked: applied_filters[:recruitment_cycle_year]&.include?(year),
+          }
```

Comment:
Could we loop through visible years instead to avoid the select + map?
```
RecruitmentCycle.years_visible_to_providers.map do |year|
  {
    value: year.to_s,
    label: RecruitmentCycle::CYCLES[year.to_s],
    checked: blah,
  }
end

Reply:
Shorter now but much insistence on `to_s` needed.